db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1879,6 +1879,20 @@ Status DBImpl::RunManualCompaction(
       assert(!exclusive || !manual_conflict);
       // Running either this or some other manual compaction
       bg_cv_.Wait();
+      if (manual_compaction_paused_ > 0 && !manual.done &&
+          !manual.in_progress) {
+        manual.done = true;
+        manual.status =
+            Status::Incomplete(Status::SubCode::kManualCompactionPaused);
+        if (ca && ca->prepicked_compaction) {
+          ca->prepicked_compaction->is_canceled = true;
+          if (ca->prepicked_compaction->compaction) {
+            ca->prepicked_compaction->compaction->ReleaseCompactionFiles(
+                manual.status);
```

Comment:
I can see the desire to release files as soon as possible, but would it also work to let the cleanup happen eventually when the background compaction finally runs? It kind of worries me to have a special foreground code path for releasing input files.

Reply:
I think it makes sense to have the cleanup always happen in the background to avoid the cleanup conflict between background and foreground. (so technically we could remove the `in_progress` check above)